Backpack/Dialog/Classes/BPKDialogView.m

```diff
@@ -177,16 +171,52 @@ - (void)setupConstraints {
     ]];
 }
 
+- (void)setupIconView {
+    if (self.hasIcon && self.iconView == nil) {
+        self.iconView = [BPKDialogIconView new];
+        [self addSubview:self.iconView];
+        self.iconView.iconDefinition = self.iconDefinition;
+
+        CGSize iconViewSize = [[self.iconView class] viewSize];
+        [NSLayoutConstraint activateConstraints:@[
+            [self.iconView.topAnchor constraintEqualToAnchor:self.iconView.superview.topAnchor],
+            [self.iconView.centerXAnchor constraintEqualToAnchor:self.contentView.centerXAnchor],
+
+            [self.contentView.topAnchor constraintGreaterThanOrEqualToAnchor:self.iconView.bottomAnchor
+                                                                    constant:-(iconViewSize.height / 2.0)],
+
+            // NOTE: The constant here is explicitly not BPKSpacingLg because the icon view has a white border which
+            // blends with the contentView background. Using BPKSpacingLg makes it look unaligned with the BPKSpacingLg
+            // below. The border is BPKSpacingMd wide which when added to BPKSpacingBase makes it BPKSpacingLg. This
+            // ligns up nicely with the BPKSpacingLg between descriptionLabel and buttonStackView
+            [self.titleLabel.topAnchor constraintGreaterThanOrEqualToAnchor:self.iconView.bottomAnchor
+                                                                   constant:BPKSpacingBase]
+        ]];
+    }
+
+    if (!self.hasIcon && self.iconView != nil) {
+        [self.iconView removeFromSuperview];
```

Comment:
This is a bit surprising in a method called `setupIconView` since it does the opposite of setting up